File: ChatGPT/Desarrollo Web Endpoint RESTful/golang/endpoint.go
package main
import (
    "github.com/gin-gonic/gin"
)
func main() {
    r := gin.Default()
    r.GET("/saludo", func(c *gin.Context) {
        nombre := c.Query("nombre")
        c.JSON(200, gin.H{"mensaje": "Hola, " + nombre})
    })
    r.Run()
}

// Prueba con:
// curl "http://localhost:8080/saludo?nombre=Juan"

File: ChatGPT/Desarrollo Web Endpoint RESTful/ruby/endpoint.rb
require 'sinatra'
require 'json'
get '/saludo' do
  nombre = params['nombre'] || ''
  content_type :json
  { mensaje: "Hola, #{nombre}" }.to_json
end

# Prueba con:
# curl "http://localhost:4567/saludo?nombre=Juan"
